app/components/fathom.tsx

```diff
@@ -0,0 +1,26 @@
+'use client';
+import { load, trackPageview } from 'fathom-client';
+import { usePathname, useSearchParams } from 'next/navigation';
+import { useEffect } from 'react';
+
+export default function Fathom() {
```

Comment:
If this is being applied to every page, IMO it should just be a const inside of `layout.tsx`, rather than having its own file in `components`. And in this case, it doesn't even render anything - this `useEffect` should just be in `layout.tsx`

Reply:
Can't do that because `layout.tsx` is a server component and using `useEffect` is allowed only for client component. If I make the `layout.tsx` file a client component then exporting meta data is not allowed.